src/main.go

```diff
@@ -276,51 +285,63 @@ func garbageClean(directory, cname, current_version string, size_wanted int64) e
 }
 
 func main() {
-	repo_url := flag.String("repo", "ghcr.io/gardenlinux/gl-oci", "OCI repository to download from")
-	media_type := flag.String("media-type", "application/io.gardenlinux.uki", "artifact media type to fetch")
-	target_dir := flag.String("target-dir", "/efi/EFI/Linux", "directory to write artifacts to")
-	os_release_path := flag.String("os-release", "/etc/os-release", "alternative path where the os-release file is read from")
-	skip_efi_check := flag.Bool("skip-efi-check", false, "skip performing EFI safety checks")
+	flag_set := flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+	flag_set.SetOutput(os.Stderr)
+
+	repo_url := flag_set.String("repo", "ghcr.io/gardenlinux/gl-oci", "OCI repository to download from")
+	media_type := flag_set.String("media-type", "application/io.gardenlinux.uki", "artifact media type to fetch")
+	target_dir := flag_set.String("target-dir", "/efi/EFI/Linux", "directory to write artifacts to")
+	os_release_path := flag_set.String("os-release", "/etc/os-release", "alternative path where the os-release file is read from")
+	skip_efi_check := flag_set.Bool("skip-efi-check", false, "skip performing EFI safety checks")
 
-	flag.Usage = func() {
-		fmt.Fprintf(flag.CommandLine.Output(), "Usage: %s [options] <version>\n", os.Args[0])
-		flag.PrintDefaults()
+	flag_set.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [options] <version>\n\n", os.Args[0])
+		flag_set.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "\nexit codes:\n\t0: success\n\t1: invalid arguments\n\t2: system failure\n\t3: network problems\n")
 	}
 
-	flag.Parse()
-	if flag.NArg() < 1 {
-		fmt.Println("Error: version argument is required")
+	if err := flag_set.Parse(os.Args[1:]); err != nil {
 		os.Exit(1)
 	}
-	version := flag.Arg(0)
+
+	if flag_set.NArg() < 1 {
+		fmt.Fprintln(os.Stderr, "Error: version argument is required")
+		os.Exit(1)
+	}
+	version := flag_set.Arg(0)
 
 	cname, current_version, err := getCname(*os_release_path)
 	if err != nil {
-		panic(err)
+		fmt.Fprintln(os.Stderr, "Error:", err)
+		os.Exit(1)
```

Comment:
maybe use constants instead of the int literals for easier reading?

`return(ERR_INVALID_ARGUMENTS)`